lib/iris/_lazy_data.py

```diff
@@ -158,3 +158,18 @@ def multidim_lazy_stack(stack):
         result = da.stack([multidim_lazy_stack(subarray)
                            for subarray in stack])
     return result
+
+
+def co_realise_cubes(cubes):
+    """
+    Fetch real data for multiple cubes at one time.
+
+    This fetches lazy content, equivalent to accessing each cube.data.
+    However, lazy calculations and data fetches can be shared between the
+    calculations, improving performance.
+
+    """
+    results = da.compute(list(cube.core_data() for cube in cubes))
```

Comment:
In `as_concrete_data()` there's an instance check for a `MaskedConstant`. This is needed because the dtype of a `MaskedConstant` is always float, so we create a scalar masked array with the correct dtype instead. You probably want to do the same here. In fact, it might be worth extending `as_concrete_data()` to work with sequences, and calling it here, to prevent duplication of code.

Reply:
Very good spot ! :bouquet: 
And a good suggestion.  Will look into it...